addons/website_sale_loyalty/controllers/portal.py

```diff
@@ -0,0 +1,36 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.tools.misc import format_amount
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+
+class CustomerPortalLoyalty(CustomerPortal):
+
+    @route()
+    def portal_get_card_history_values(self, card_id):
+        """
+        Retrieve card history values and include published trigger products for the loyalty
+        program.
```

Comment:
Now here it's the other way around, this function is responsible for adding trigger products but the ovveride does not retrieve the card history values.
```suggestion
        Add published trigger products for the loyalty
        program.
```